examples/GPU_Examples/SelfCalibrationReconstruction_NonCartesian_gpuNUFFT_SENSE.py

```diff
@@ -0,0 +1,123 @@
+"""
+Non Cartesian Self Calibrating Reconstruction
+====----=====================================
+
+Author: Chaithya G R
+
+In this tutorial we will reconstruct an MRI image from non cartesian kspace
+measurements.
+
+Import neuroimaging data
+------------------------
+
+We use the toy datasets available in pysap, more specifically a 2D parallel MRI
+brain slice on 32 channels and the acquisition non cartesian scheme.
+"""
+
+# Package import
+from mri.operators import NonCartesianFFT, WaveletN
+from mri.reconstructors import SelfCalibrationReconstructor
+from mri.reconstructors.utils.extract_sensitivity_maps import get_Smaps
+from mri.operators.utils import convert_locations_to_mask, \
+    gridded_inverse_fourier_transform_nd
+import pysap
+from pysap.data import get_sample_data
+
+# Third party import
+from modopt.math.metrics import ssim
+from modopt.opt.linear import Identity
+from modopt.opt.proximity import SparseThreshold
+import numpy as np
+
+# Loading input data
+cartesian_ref_image = get_sample_data('2d-pmri')
+image = pysap.Image(data=np.sqrt(np.sum(cartesian_ref_image.data**2, axis=0)))
+
+# Obtain MRI cartesian mask
+mask = get_sample_data("mri-radial-samples")
+kspace_loc = mask.data
+
+# View Input
+# image.show()
+# mask.show()
+
+#############################################################################
+# Generate the kspace
+# -------------------
+#
+# From the 2D brain slice and the acquisition mask, we retrospectively
+# undersample the k-space using a cartesian acquisition mask
+# We then reconstruct the zero order solution as a baseline
+
+# Get the locations of the kspace samples and the associated observations
+fourier_op = NonCartesianFFT(
+    samples=kspace_loc,
+    shape=image.shape,
+    n_coils=cartesian_ref_image.shape[0],
+    implementation='gpuNUFFT'
+)
+kspace_obs = fourier_op.op(cartesian_ref_image)
+
+# Gridded solution
+grid_space = np.linspace(-0.5, 0.5, num=image.shape[0])
+grid2D = np.meshgrid(grid_space, grid_space)
+grid_soln = np.asarray([
+    gridded_inverse_fourier_transform_nd(kspace_loc, kspace_obs_ch,
+                                         tuple(grid2D), 'linear')
+    for kspace_obs_ch in kspace_obs
+])
+image_rec0 = pysap.Image(data=np.sqrt(np.sum(np.abs(grid_soln)**2, axis=0)))
+# image_rec0.show()
+base_ssim = ssim(image_rec0, image)
+print('The Base SSIM is : ' + str(base_ssim))
+
+# Obtain SMaps
+Smaps, SOS = get_Smaps(
+    k_space=kspace_obs,
+    img_shape=fourier_op.shape,
+    samples=kspace_loc,
+    thresh=(0.01, 0.01),
```

Comment:
I realize that thresh is not necessarily a very good variable name, so maybe a comment here would not be too much to explain what this does.